exchange/src/exchange/wrappers/handle/wrapper_handle.cpp

```diff
@@ -82,32 +107,32 @@ WrapperHandle::WrapperHandle(const std::string& algo_path) :
 void
 WrapperHandle::block_on_init()
 {
-    while (!std::holds_alternative<common::init_message>(reader_.get_message())) {}
+    while (!std::holds_alternative<common::init_message>(reader_->get_message())) {}
 }
 
 WrapperHandle::WrapperHandle(
     const std::vector<std::string>& args, const std::string& algorithm
-)
+) : reader_(std::make_unique<PipeReader>()), writer_(std::make_unique<PipeWriter>())
 {
     static const std::string path{wrapper_binary_path()};
 
-    auto& pipe_in_ptr = reader_.get_pipe();
-    auto& pipe_out_ptr = writer_.get_pipe();
+    auto& pipe_in_ptr = reader_->get_pipe();
+    auto& pipe_out_ptr = writer_->get_pipe();
 
     wrapper_ = bp::child(
         bp::exe(path), bp::args(args), bp::std_in<pipe_out_ptr, bp::std_err> stderr,
         bp::std_out > pipe_in_ptr
     );
 
     using algorithm_t = nutc::common::algorithm_content;
-    algorithm_t algorithm_message = algorithm_t(algorithm);
+    algorithm_t algorithm_message = algorithm_t(common::base64_encode(algorithm));
```

Comment:
Just to confirm - we're always encoding the algo (including py)? I'm okay with this, just wanna confirm